<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>RNAlib-2.4.6: Changelog</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() {
    if ($('.searchresults').length > 0) { searchBox.DOMSearchField().focus(); }
  });
</script>
<link rel="search" href="search_opensearch.php?v=opensearch.xml" type="application/opensearchdescription+xml" title="RNAlib-2.4.6"/>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">RNAlib-2.4.6
   </div>
  </td>
   <td>        <div id="MSearchBox" class="MSearchBoxInactive">
          <div class="left">
            <form id="FSearchBox" action="search.html" method="get">
              <img id="MSearchSelect" src="search/mag.png" alt=""/>
              <input type="text" id="MSearchField" name="query" value="Search" size="20" accesskey="S" 
                     onfocus="searchBox.OnSearchFieldFocus(true)" 
                     onblur="searchBox.OnSearchFieldFocus(false)"/>
            </form>
          </div><div class="right"></div>
        </div>
</td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('changelog.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="headertitle">
<div class="title">Changelog </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>v2.4.6</p><ul>
<li>Stabilize rounding of free energy output in RNAalifold</li>
<li>API: Fix potential rounding errors for comparative free energies in eval.c and mfe.c</li>
<li>API: Fix regression in exterior loop dangling end contributions for comparative base pair probabilities and Boltzmann sampling (introduced with v2.4.4)</li>
<li>API: Fix regression with hard constrained base pairs for comparative structure prediction (introduced with v2.4.4)</li>
<li>TESTS: Add basic tests for RNAalifold executable</li>
<li>TESTS: Ignore 'frequency of MFE structure' in RNAcofold partition function checks</li>
</ul>
v2.4.5</p><ul>
<li>Allow for arbitrary number of input files in RNAfold</li>
<li>Allow for parallel processing of input data in RNAfold (UNIX only, no Windows support yet)</li>
<li>Add SHAPE reactivity support through commandline options for RNAplfold</li>
<li>Fix unstructured domain motif detection in MFE, centroid, and MEA structures computed by RNAfold</li>
<li>Limit allowed set of commands in command file for RNAcofold to hard and soft constraints</li>
<li>API: Add functions to compute equilibrium probability of particular secondary structures</li>
<li>API: Add dynamic string stream data type and associated functions</li>
<li>API: Add priority-queue like data structure with unordered fill capability and ordered output callback execution</li>
<li>API: Add functions to detect unstructured domain motifs in MFE, centroid, and MEA structures</li>
<li>API: Fix bug in sliding-window partition function computation with SHAPE reactivity and Deigan et al. conversion method</li>
<li>API: Fix application of '&lt;' and '&gt;' constraint symbols in dot-bracket provided constraints (was broken since v2.4.2)</li>
<li>API: Fix MEA structure computation in the presence of unstructured domains</li>
<li>API: Stabilize order of probability entries in EPS dot-plot files</li>
<li>Fix compiler warnings on wrong type of printf() in naview.c</li>
<li>Define VRNA_VERSION macro as string literal and add macros for major, minor, and patch numbers</li>
<li>Stabilize parallel make of Mac OS X installer</li>
<li>Add energy parameter set from Langdon et al. 2018</li>
<li>Add autoconf checks for POSIX threads compiler/linker support</li>
<li>SWIG: Fix 'next' is a perl keyword warnings for Perl5 wrapper</li>
<li>SWIG: Catch errors and throw execptions whenever scripting language provided callback functions are not applicable or fail</li>
<li>SWIG: Add keyword arguments and autodoc feature for Python/Python3 wrappers</li>
</ul>
v2.4.4</p><ul>
<li>Change verbose output for soft-constraints derived ligand binding motifs in RNAfold</li>
<li>Allow for lowercase letters in ct2db input</li>
<li>Fix bug in interior-like G-Quadruplex MFE computation for single sequences</li>
<li>Fix autoconf switch to enable deprecation warnings</li>
<li>Fix bug in eval_int_loop() that prevented propagation of energy evaluation for loops with nick in strands</li>
<li>Fix several bugs for SHAPE reactivity related comparative partition function computations</li>
<li>Fix annotation of PostScript output for soft-constraint derived ligand binding motifs in RNAfold</li>
<li>Fix constraint indices for multibranch loops in unpaired probability computations of LPfold.c</li>
<li>Fix dangling end contributions in comparative partition function for exterior loops</li>
<li>API: Add simplified interface for <a class="el" href="group__pf__cofold.html#ga4e5c7d06c302a7c59fc0d64dc142ca63" title="Calculate partition function and base pair probabilities of nucleic acid/nucleic acid dimers...">vrna_pf_dimer()</a></li>
<li>API: Move concentraton dependent implementation for co-folding to separate compile unit</li>
<li>API: Add new API functions for exterior loop evaluations</li>
<li>API: Add simplified interfaces for energy evaluation with G-Quadruplexes and circular RNAs</li>
<li>API: Add findpath functions that allow for specification of an upper bound for the saddle point</li>
<li>Add configure-time linker check for Python3 interface</li>
<li>Add automatic CPP suggestions for deprecated function substitutes</li>
<li>Major restucturing and constraints feature additions in loop type dependent energy evaluation functions</li>
<li>Major restructuring in MFE implementations</li>
<li>Major restructuring in PF implementations</li>
<li>Minor fixes in Boltzmann sampling implementation</li>
<li>SWIG: Fix wrappers for findpath() implementation</li>
<li>SWIG: Add tons of energy evaluation wrappers</li>
<li>SWIG: Fix configure-time check of Perl5 interface build capabilities</li>
<li>SWIG: Wrap functions from walk.c and neighbor.c</li>
<li>DOC: Add some missing references to manpages of executable programs</li>
<li>REFMAN: Heavy re-ordering of the RNAlib reference manual</li>
</ul>
v2.4.3</p><ul>
<li>Fix handling of dangling end contribution at sequence boundaries for sliding window base pair probability computations</li>
<li>Fix handling of base pair hard constraints in sliding-window implementations</li>
<li>Fix sliding-window pair probability computations with multibranch-loop unpaired constraints</li>
<li>Fix sliding-window non-specific base pair hard constraint implementation</li>
<li>Fix probability computation for stochastic backtracking in RNAsubopt --stochBT_en output</li>
<li>Fix regression in comparative structure prediction for circular RNAs</li>
<li>Fix LDFLAGS for scripting language interfaces in corresponding Makefiles</li>
<li>Stabilize partition function scaling by always using sfact scaling factor from model details</li>
<li>Add --pf_scale commandling parameter to RNAplfold</li>
<li>Add constraint framework for single sequence circular RNA structure prediction</li>
<li>Add RNAfold test suite to check for working implementation of constraints for circular RNAs</li>
<li>Add a brief contribution guideline CONTRIBUTING.md</li>
<li>Prevent RNAplfold from creating inf/-inf output when solution set is empty with particular hard constraints</li>
<li>Include RNAforester v2.0.1</li>
</ul>
v2.4.2</p><ul>
<li>Fix G-Quadruplex energy corrections in comparative structure energy evaluations</li>
<li>Fix discrepancy in comparative exterior loop dangling end contribution of eval vs. MFE predictions</li>
<li>Fix regression in RNAup unstructuredness and interaction energy computations</li>
<li>Fix sequence length confusions when FASTA input contains carriage returns</li>
<li>Fix build problems of RNAlocmin with older compilers</li>
<li>Fix sliding-window hard constraints where single nucleotides are prohibited from pairing</li>
<li>Fix dot-bracket output string length in sliding-window MFE with G-Quadruplexes</li>
<li>Fix unpaired probability computations for separate individual loop types in LPfold.c</li>
<li>Fix bad memory access in RNAsubopt with dot-bracket constraint</li>
<li>Add full WUSS support for --SS_cons constraint option in RNAalifold</li>
<li>Add commandline option to RNALalifold that enables splitting of energy contributions into separate parts</li>
<li>Add missing hard constraint cases to sliding-window partition function implementation</li>
<li>Add CSV output option to RNAcofold</li>
<li>Use the same model details for SCI computations in RNAalifold</li>
<li>Abort computations in <a class="el" href="group__eval.html#gab12e6b1226227670322150df018734f8" title="Calculate the free energy of an already folded RNA and print contributions on a per-loop base...">vrna_eval_structure_v()</a> if structure has unexpected length</li>
<li>Use original MSA in all output generated by RNAalifold and RNALalifold</li>
<li>API: Add new functions to convert dot-bracket like structure annotations</li>
<li>API: Add various new utility functions for alignment handling and comparative structure predictions</li>
<li>API: Add function <a class="el" href="group__string__utils.html#gac632775617f4b2df096bfe8efb53d07f" title="Split a string into tokens using a delimiting character. ">vrna_strsplit()</a> to split string into tokens</li>
<li>API: Do not convert sequences of input MSA to uppercase letters in <a class="el" href="group__file__utils.html#ga59204cd1daa4927f5127cc65a2886efd" title="Read a multiple sequence alignment from file handle. ">vrna_file_msa_read_record()</a></li>
<li>API: Rename vrna_annotate_bp_covar() and vrna_annotate_pr_covar()</li>
<li>API: Add new noLP neighbor generation</li>
<li>SWIG: Add wrapper for functions in file_utils_msa.h</li>
<li>SWIG: Add wrappers for <a class="el" href="group__subopt__stochbt.html#ga901fe42a33b07be083421741bf7dc610" title="Sample a secondary structure (consensus structure) from the Boltzmann ensemble according its probabil...">vrna_pbacktrack()</a> and <a class="el" href="group__subopt__stochbt.html#ga347375a4da8bdff74639529847f97d4f" title="Sample a secondary structure of a subsequence from the Boltzmann ensemble according its probability...">vrna_pbacktrack5()</a></li>
<li>SWIG: Add vrna_db_to_element_string() to scripting language interface</li>
<li>REFMAN: Fix formula to image conversion in HTML output</li>
</ul>
v2.4.1</p><ul>
<li>Fix memory leak in fold_compound methods of SWIG interface</li>
<li>Fix memory leaks in double ** returning functions of SWIG Perl5 interface</li>
<li>Fix memory leak in vrna_ep_t to-string() function of SWIG interface</li>
<li>Regression: Fix reverting pf_scale to defaults after <a class="el" href="group__energy__parameters.html#gad607bc3a5b5da16400e2ca4ea5560233" title="Rescale Boltzmann factors for partition function computations. ">vrna_exp_params_rescale()</a></li>
<li>Regression: Fix homo-dimer partition function computation in RNAcofold</li>
<li>Add unit tests for RNAcofold executable</li>
<li>Add SHAPE reactivity support to RNAcofold</li>
<li>Add SHAPE reactivity support to RNALalifold</li>
</ul>
v2.4.0</p><ul>
<li>Bump libsvm to version 3.22</li>
<li>Print G-Quadruplex corrections in verbose mode of RNAeval</li>
<li>Change behavior of RNAfold --outfile option to something more predictable</li>
<li>Unify max_bp_span usage among sliding window prediction algorithms: RNAplfold, RNALfold, and RNALalifold now consider any base pair (i,j) with (j - i + 1) &lt;= max_bp_span</li>
<li>Add SHAPE reactivity data support to RNALfold</li>
<li>Add commands-file support for RNALfold, RNAplfold (hard/soft constraints)</li>
<li>Add RNAlocmin - Calculate local minima from structures via gradient walks</li>
<li>Add RNA Bioinformatics tutorial (PDF version)</li>
<li>Add hard constraints to sliding-window MFE implementations (RNALfold, RNALalifold)</li>
<li>Add hard constraints to sliding-window PF implementations (RNAplfold)</li>
<li>Add soft constraints to sliding-window MFE implementation for single sequences (RNALfold)</li>
<li>Add soft constraints to sliding-window PF implementations (RNAplfold)</li>
<li>Add SWIG interfaces for sliding-window MFE/PF implementations</li>
<li>Add proper SWIG interface for alignment and structure plotting functions</li>
<li>Add proper SWIG interface for duplexfold, duplex_subopt, and its comparative variants</li>
<li>Add SWIG wrapper for <a class="el" href="group__energy__parameters.html#gad607bc3a5b5da16400e2ca4ea5560233" title="Rescale Boltzmann factors for partition function computations. ">vrna_exp_params_rescale()</a></li>
<li>Add explicit destructor for SWIG generated vrna_md_t objects</li>
<li>Add SWIG perl5 typemap for simple nested STL vectors</li>
<li>Add dummy field in <a class="el" href="structvrna__structured__domains__s.html">vrna_structured_domains_s</a></li>
<li>Add note about SSE optimized code in reference manual</li>
<li>Add SWIG interface for findpath implementation</li>
<li>Add prepare() functions for ptypes-arrays and vrna_(exp_)param_t</li>
<li>Add warnings for ignored commands in function <a class="el" href="group__file__utils.html#ga5e993fc4b9602af73aaaab4d3b3cd9a9" title="Apply a list of commands to a vrna_fold_compound_t. ">vrna_commands_apply()</a></li>
<li>Add callback featured functions for sliding window MFE and PF implementations</li>
<li>Change default behavior of adding soft constraints to a vrna_fold_compound_t (store only)</li>
<li>Several fixes with respect to G-Quadruplex prediction in sliding-window MFE recursions (single sequence and comparative implementation)</li>
<li>Replace comparative sliding-window MFE recursions (All hits are reported to callback and can be filtered in a post-processing step)</li>
<li>API: Remove E_mb_loop_stack() and introduce new function <a class="el" href="group__loops.html#ga73f048d35475e1d8f50b7d4ac6a2aeb6" title="Evaluate energy of a multi branch helices stacking onto closing pair (i,j) ">vrna_E_mb_loop_stack()</a> as a replacement</li>
<li>API: change data type of all constraint bit-flags from `char` to `unsigned char`</li>
<li>API: change data type of a2s array in comparative structure prediction from `unsigned short` to `unsigned int`</li>
<li>API: Change function parameter order in <a class="el" href="group__local__pf__fold.html#ga70ec33810fa17a9f67a7c59f156d0137" title="Compute various equilibrium probabilities under a sliding window approach. ">vrna_probs_window()</a> to follow the style of other callback-aware functions in RNAlib</li>
<li>Move sliding-window MFE implementations to new file mfe_window.c</li>
<li>Fix building PDF Reference manual with non-standard executable paths</li>
<li>Fix redefinition of macro ON_SAME_STRAND() in subopt.c</li>
<li>Fix dangling end issues in sliding-window MFE implementations</li>
<li>Fix regression for --canonicalBPonly switch in RNAfold/RNAcofold/RNAsubopt</li>
<li>Fix building sliding-window MFE implementation without SVM support</li>
<li>Fix parsing of STOCKHOLM 1.0 MSA files that contain MSA spanning multiple blocks</li>
<li>Fix Alidot link in RNAalifold manpage</li>
<li>Fix wrong pre-processor flags when enabling single-precision PF computations</li>
<li>Fix unit testing perl5 interface by including builddir/tests in PERL5LIB path</li>
<li>Fix buffer overflow in hairpin loop sequence motif extraction for circular RNAs</li>
<li>Fix out-of-bounds memory access in neighbor.c</li>
<li>Restore capability to compile stand-alone findpath utility</li>
<li>Restore capability to use non-standard alphabets for structure prediction</li>
<li>Restore old-API random number functions in SWIG interface</li>
<li>Allow additional control characters in MAF MSA input that do not end a block</li>
<li>Improve reference manual</li>
<li>Make functions in <a class="el" href="pair__mat_8h_source.html">pair_mat.h</a> static inline</li>
<li>Prevent users from adding out-of-range base pair soft constraints</li>
<li>Inline print functions in color_output.inc</li>
<li>Start documenting callback features in reference manual</li>
<li>Re-write large portions of sliding-window PF implementation</li>
<li>Introduce soft-constraint state flag</li>
<li>Clean-up SWIG unit test framework</li>
<li>Remove obsolete scripts ct2b.pl and colorrna.pl from src/Utils directory</li>
<li>Remove old RNAfold tutorial</li>
</ul>
v2.3.5</p><ul>
<li>Fix duplication of output filename prefix in RNAfold</li>
<li>Add V3.0 API for sliding window partition function (a.k.a. RNAPLfold)</li>
<li>Add G-Quadruplex prediction to RNALalifold</li>
<li>Add SWIG wrappers for callback-based sliding window comparative MFE prediction</li>
<li>Add SSE4.1 multiloop decomposition for single sequence MFE prediction</li>
<li>Enable RNAfold unit tests to run in paralllel</li>
<li>Enable users to turn-off base pair probability computations in RNAcofold with -a option</li>
<li>Split move set in neighbor.c</li>
</ul>
v2.3.4</p><ul>
<li>Fix G-Quadruplex probability computation for single sequences</li>
<li>Fix double-free when using SHAPE reactivity data in RNAalifold</li>
<li>Fix out-of-bounds access in strand_number array</li>
<li>Fix weighting of SHAPE reactivity data in consensus structure prediction when fewer data than sequences are present</li>
<li>Fix z-score output in RNALfold</li>
<li>Substitute field name 'A0'/'B0' in data structure <a class="el" href="structvrna__dimer__conc__s.html" title="Data structure for concentration dependency computations. ">vrna_dimer_conc_s</a> by 'Ac_start'/'Bc_start' to avoid clashes with termios.h (Mac OSX Python wrapper bug)</li>
<li>Minimize usage of 'unsafe' sprintf() calls</li>
<li>Enhance auto-id feature in executable programs</li>
<li>Always sanitize output file names to avoid problems due to strange FASTA headers</li>
<li>Lift restrictions of FASTA header length in RNAfold, RNAcofold, and RNAeval</li>
<li>Add ViennaRNA/config.h with pre-processor definitions of configure time choices</li>
<li>Add test-suite for RNAfold</li>
<li>Add functions to procude colored EPS structure alignments</li>
<li>Add function to write Stockholm 1.0 formatted alignments</li>
<li>Add function to sanitize file names</li>
<li>Add callback based implementation for sliding-window MFE prediction (single sequences, comparative structure prediction)</li>
<li>Add fast API 3.0 implementations to generate structural neighbors and perform steepest descent / random walks (Thanks to Gregor!)</li>
<li>Add parameter option to RNALalifold for colored EPS structure alignment and structure plot output</li>
<li>Add parameter option to RNALalifold to write hits into Stockholm file</li>
<li>Add parameter option to RNAalifold to write Stockholm 1.0 formatted output</li>
<li>Add parameter option to RNAalifold to suppress stderr spam</li>
<li>Add auto-id feature to RNAplot, RNALfold, RNAsubopt, RNAplfold, RNAheat</li>
<li>Add SHAPE reactivity derived pseudo-energies as separate output in RNAalifold</li>
<li>Add colored output to RNA2Dfold, RNALalifold, RNALfold, RNAduplex, RNAheat, RNAinverse, RNAplfold, and RNAsubopt</li>
<li>Add command line parameters to RNAsubopt to allow for specification of input/output files</li>
</ul>
v2.3.3</p><ul>
<li>Fix multiloop contributions for comparative partition function</li>
<li>Fix building python2 extension module for OSX</li>
</ul>
v2.3.2</p><ul>
<li>Fix pair probability plist creation with G-Quadruplexes</li>
<li>Allow for specification of python2/3-config at configure time</li>
<li>Fix init of vrna_md_t data structure after call to <a class="el" href="group__model__details.html#gabad896c3650d420f3f3ddefc69e2bceb" title="Set default model details. ">set_model_details()</a></li>
<li>Fix bug in consensus partition function with hard constraints that force nucleotides to be paired</li>
<li>Fix compilation of functions that use ellipsis/va_list</li>
<li>Enable generic hard constraints by default</li>
<li>Fix init of partition function DP matrices for unusually short RNAs</li>
<li>Fix behavior of RNAplfold for unusually short RNAs</li>
<li>Report SCI of 0 in RNAalifold when sum of single sequence MFEs is 0</li>
<li>Avoid multiple includes of <a class="el" href="pair__mat_8h_source.html">pair_mat.h</a></li>
<li>Add configure flag to build entirely static executables</li>
</ul>
v2.3.1</p><ul>
<li>Add description for how to use unstructured domains through command files to reference manual and RNAfold manpage</li>
<li>Fix compilation issue for Windows platforms with MingW</li>
<li>Add missing newline in non-TTY-color output of <a class="el" href="group__utils.html#ga039bae6153a6415b054dbe6045f83d03" title="Print an info message. ">vrna_message_info()</a></li>
<li>Fix regression in <a class="el" href="group__model__details.html#ga36ae40b8c3b82362f5798ad5b047b814" title="Update the model details data structure. ">vrna_md_update()</a> that resulted in incomplete init of reverse-basepair type array</li>
<li>Extend coverage of generic hard constraints for partition function computations</li>
<li>Fix scaling of secondary structure in EPS plot such that it always fits into bounding box</li>
<li>Several fixes and improvements for SWIG generated scripting language interface(s)</li>
</ul>
v2.3.0</p><ul>
<li>Add grammar extension with structured and unstructured domains</li>
<li>Add default implementation for unstructured domains to allow for ligand/protein binding to unpaired structure segments (MFE and PF for single sequences)</li>
<li>Introduced command files that subsume constraint definition files (currently used in RNAfold and RNAcofold)</li>
<li>Replace explicit calls to asprintf() with portable equivalent functions in the library</li>
<li>Fix configure script to deal with situations where Perl module can't be build</li>
<li>Fix bug in doc/Makefile.am that prevented HTML installation due to long argument list</li>
<li>Added utility functions that deal with conversion between different units</li>
<li>Bugfix in SWIG wrapped generic soft constraint feature</li>
<li>Add <a class="el" href="group__subopt__wuchty.html#ga700f662506a233e42dd7fda74fafd40e" title="Returns list of subopt structures or writes to fp. ">subopt()</a> and subopt_zuker() methods to SWIG wrapped fold_compound objects</li>
<li>Bugfix multiloop decomposition in MFE for circular RNAs</li>
<li>Add separate function to compute pscore for alignments</li>
<li>Renamed VRNA_VC_TYPE_* macros to VRNA_FC_TYPE_*</li>
<li>Bugfix regression that prevented programs to fail on too long input sequences</li>
<li>Extend EPS dot-plot in RNAfold to include motif/binding probabilities from unstructured domains</li>
<li>Add variadic functions for error/warning/info message</li>
<li>Add ID manipulation feature to RNAeval</li>
<li>Extend API for soft constraint feature for more fine-grained control</li>
<li>Add section on SWIG wrapped functions in reference manual</li>
<li>Fix bug in interior loop computations when hard constraints result in non-canonical base pairs</li>
</ul>
v 2.2.10</p><ul>
<li>Do not 'forget' subopt results when output is not written to file handle and sorting is switched off</li>
<li>Fix bad memory access in <a class="el" href="group__subopt__wuchty.html#ga0f11d738fb8c8b1885a90c11c8931ff6" title="Returns list of subopt structures or writes to fp. ">vrna_subopt()</a> with sorted output</li>
<li>Add SWIG wrappers for <a class="el" href="group__subopt__wuchty.html#ga1053837e6b6f158093508f8a70998352" title="Generate suboptimal structures within an energy band arround the MFE. ">vrna_subopt_cb()</a></li>
<li>Correctly show if C11 features are activated in configure status</li>
<li>Fix autoconf checks to allow for cross compilation again</li>
</ul>
v 2.2.9</p><ul>
<li>Fix bug in partition function scaling for backward compatibility of ali_pf_fold()</li>
<li>Stabilize v3.0 API when building RNAlib and third party program linking against it with compilers that use different C/C++ standards</li>
<li>Add details on how to link against RNAlib to the reference manual</li>
<li>Fix RNAlib2.pc</li>
<li>Fix bug for temperature setting in RNAplfold</li>
<li>Use -fflat-lto-objects for static RNAlib library to allow linking without LTO</li>
<li>Fix interpretation of 'P' hard constraint for single nucleotides in constraint definition files</li>
<li>Add 'A' command for hard constraints</li>
<li>Fix several hard constraint corner-cases in MFE and partition function computation when nucleotides must not be unpaired</li>
<li>Fix order of hard constraints when read from input file</li>
<li>Allow for non-canonical base pairs in MFE and partition function computations if hard constraints demand it</li>
<li>Fix behavior of --without-swig configure script option</li>
<li>Fix bug in hard constraints usage of exterior loop MFE prediction with odd dangles</li>
<li>Add parsers for Clustal, Stockholm, FASTA, and MAF formatted alignment files</li>
<li>Enable RNAalifold to use Clustal, Stockholm, FASTA, or MAF alignments as input</li>
<li>Lift restriction of sequence number in alignments for RNAalifold</li>
<li>Enable ANSI colors for TTY output in RNAfold, RNAcofold, RNAalifold, RNAsubopt, and warnings/errors issued by RNAlib</li>
<li>Add various new commandline options to manipulate sequence/alignment IDs in RNAfold, RNAcofold and RNAalifold</li>
</ul>
v 2.2.8</p><ul>
<li>Fix bad memory access in RNAalifold</li>
<li>Fix regression in RNAalifold to restore covariance contribution ratio determination for circular RNA alignments</li>
<li>Changed output of RNAsubopt in energy-band enumeration mode to print MFE and energy range in kcal/mol instead of 10cal/mol</li>
<li>Include latest Kinfold sources that make use of v3.0 API, therefore speeding up runtime substantially</li>
<li>Re-activate warnings in RNAeval when non-canonical base pairs are encountered</li>
<li>Fix syntactic incompatibilities that potentially prevented compilation with compilers other than gcc</li>
<li>dd function to compare nucleotides encoded in IUPAC format</li>
<li>Fix regression in energy evaluation for circular RNA sequences</li>
<li>Fix regression in suboptimal structure enumeration for circular RNAs</li>
<li>Allow for P i-j k-l commands in constraint definition files</li>
<li>Make free energy evaluation functions polymorphic</li>
<li>Add free energy evaluation functions that allow for specifying verbosity level</li>
<li>Secure functions in alphabet.c against NULL pointer arguments</li>
<li>Fix incompatibility with swig &gt;= 3.0.9</li>
<li>Fix memory leak in swig-generated scripting language interface(s) for user-provided target language soft-constraint callbacks</li>
<li>Expose additional functions to swig-generated scripting language interface(s)</li>
<li>Build Python3 interface by default</li>
<li>Start of more comprehensive scripting language interface documentation</li>
<li>Fix linking of python2/python3 interfaces when libpython is in non-standard directory</li>
<li>Restructured viennarna.spec for RPM based distributions</li>
<li>Several syntactic changes in the implementation to minimize compiler warnings</li>
<li>Fix --with-*/--without-* and --enable-*/--disable-* configure script behavior</li>
</ul>
v 2.2.7</p><ul>
<li>Fix partition function scaling for long sequences in RNAfold, RNAalifold, and RNAup</li>
<li>Fix backtracking issue in RNAcofold when --noLP option is activated</li>
<li>Fix hard constraints issue for circular RNAs in generating suboptimal structures</li>
<li>Rebuild reference manual only when actually required</li>
</ul>
v 2.2.6</p><ul>
<li>Plugged memory leak in RNAcofold</li>
<li>Fixed partition function rescaling bug in RNAup</li>
<li>Fixed bug in RNALfold with window sizes larger than sequence length</li>
<li>Re-added SCI parameter for RNAalifold</li>
<li>Fixed backtracking issue for large G-quadruplexes in RNAalifold</li>
<li>Fixed missing FASTA id in RNAeval output</li>
<li>Added option to RNAalifold that allows to specify prefix for output files</li>
<li>Several fixes and additional functions/methods in scripting language interface(s)</li>
<li>Added version information for scripting language interface(s)</li>
<li>Some changes to allow for compilation with newer compilers, such as gcc 6.1</li>
</ul>
v 2.2.5</p><ul>
<li>Fixed regression in RNAcofold that prohibited output of concentration computations</li>
<li>Fixed behavior of RNAfold and RNAcofold when hard constraints create empty solution set (programs now abort with error message)</li>
<li>Added optional Python 3 interface</li>
<li>Added RNA::Params Perl 5 sub-package</li>
<li>Update RNA::Design Perl 5 sub-package</li>
<li>Simplified usage of v3.0 API with default options</li>
<li>Wrap more functions of v3.0 API in SWIG generated scripting language interfaces</li>
<li>Plugged some memory leaks in SWIG generated scripting language interfaces</li>
<li>Changed parameters of recursion status callback in vrna_fold_compound_t</li>
<li>Enable definition and binding of callback functions from within SWIG target language</li>
<li>Added optional subpackage Kinwalker</li>
<li>Added several configure options to ease building and packaging under MacOS X</li>
<li>Added new utility script RNAdesign.pl</li>
</ul>
v 2.2.4</p><ul>
<li>Fixed bug in RNAsubopt that occasionally produced cofolded structures twice</li>
<li>Removed debugging output in preparations of consensus structure prediction datastructures</li>
</ul>
v 2.2.3</p><ul>
<li>Added postscipt annotations for found ligand motifs in RNAfold</li>
<li>Added more documentation for the constraints features in RNAfold and RNAalifold</li>
<li>Restore backward compatibility of <a class="el" href="group__consensus__fold.html#ga5349960075b1847720a2e9df021e2675" title="Get pointers to (almost) all relavant arrays used in alifold&#39;s partition function computation...">get_alipf_arrays()</a></li>
</ul>
v 2.2.2</p><ul>
<li>Fix regression bug that occasionally prevented backtracking with RNAcofold --noLP</li>
</ul>
v 2.2.1</p><ul>
<li>Fix regression bug that made RNAcofold -a unusable</li>
<li>Fix regression bug that prohibited RNAfold to compute the MEA structure when G-Quadruplex support was switched on</li>
<li>Fix bug in Kinfold to enable loading energy parameters from file</li>
<li>Fix potential use of uninitialized value in RNApdist</li>
<li>Add manpage for ct2db</li>
<li>Fix MEA computation when G-Quadruplex support is activated</li>
<li>Allow for vendor installation of the perl interface using INSTALLDIRS=vendor at configure time</li>
<li>Install architecture dependent and independent files of the perl and python interface to their correct file system locations</li>
</ul>
v 2.2.0</p><ul>
<li>RNAforester is now of version 2.0</li>
<li>New program RNApvmin to compute pseudo-energy pertubation vector that minimizes discrepancy between observed and predicted pairing probabilities</li>
<li>SHAPE reactivity support for RNAfold, RNAsubopt, and RNAalifold</li>
<li>Ligand binding to hairpin- and interior-loop motif support in RNAfold</li>
<li>New commandline option to limit maximum base pair span for RNAfold, RNAsubopt, RNAcofold, and RNAalifold</li>
<li>Bugfix in RNAheat to remove numerical instabilities</li>
<li>Bugfix in RNAplex to allow for computation of interactions without length limitation</li>
<li>Bugfix in RNAplot for simple layouts and hairpins of size 0</li>
<li>(generic) hard- and soft-constraints for MFE, partition function, base pair probabilities, stochastic backtracking, and suboptimal secondary structures of single sequences, sequence alignments, and sequence dimers</li>
<li>libsvm version as required for z-scoring in RNALfold is now 3.20</li>
<li>Stochastic backtracking for single sequences is faster due to usage of Boustrophedon scheme</li>
<li>First polymorphic functions <a class="el" href="group__mfe__fold.html#gabd3b147371ccf25c577f88bbbaf159fd" title="Compute minimum free energy and an appropriate secondary structure of an RNA sequence, or RNA sequence alignment. ">vrna_mfe()</a>, <a class="el" href="group__pf__fold.html#ga29e256d688ad221b78d37f427e0e99bc" title="Compute the partition function  for a given RNA sequence, or sequence alignment. ">vrna_pf()</a>, and <a class="el" href="group__subopt__stochbt.html#ga901fe42a33b07be083421741bf7dc610" title="Sample a secondary structure (consensus structure) from the Boltzmann ensemble according its probabil...">vrna_pbacktrack()</a>.</li>
<li>The FLT_OR_DBL macro is now a typedef</li>
<li>New functions to convert between different secondary structure representations, such as helix lists, and RNAshapes abstractions</li>
<li>First object-oriented interface for new API functions in the scripting language interfaces</li>
<li>new ViennaRNA-perl submodule that augments the Perl interface to RNAlib</li>
<li>Ligand binding to hairpin- and interior-loop motif support in C-library and scripting language interfaces.</li>
<li>Libraries are generated using libtool</li>
<li>Linking of libraries and executables defaults to use Link Time Optimization (LTO)</li>
<li>Large changes in directory structure of the source code files</li>
</ul>
v 2.1.9</p><ul>
<li>Fixed integer underflow bug in RNALfold</li>
<li>Added Sequence Conservation index (SCI) option to RNAalifold</li>
<li>Fixed bug in energy evaluation of dangling ends / terminal mismatches of exterior loops and multibranch loops</li>
<li>Fixed bug in alifold partition function for circular RNAs</li>
<li>Fixed bug in alifold that scrambled backtracing with activated G-Quadruplex support</li>
<li>Fixed bug in alifold backtracking for larger G-Quadruplexes</li>
</ul>
v 2.1.8</p><ul>
<li>Repaired incorporation of RNAinverse user provided alphabet</li>
<li>Fix missing FASTA ID in RNAeval output</li>
<li>prevent race condition in parallel calls of <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a></li>
<li>Fixed memory bug in <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a> that occured using long sequences and activated G-Quad support</li>
<li>Added latest version of switch.pl</li>
</ul>
v 2.1.7</p><ul>
<li>Fixed bug in RNALfold -z</li>
<li>Python and Perl interface are compiling again under MacOSX</li>
<li>Fixed handling of C arrays in Python interface</li>
<li>Added latest version of switch.pl</li>
<li>Make relplot.pl work with RNAcofold output</li>
</ul>
v 2.1.6</p><ul>
<li>New commandline switches allow for elimination of non-canonical base pairs from constraint structures in RNAfold, RNAalifold and RNAsubopt</li>
<li>updated moveset functions</li>
<li>final fix for discrepancy of tri-loop evaluation between partition function and mfe</li>
<li>pkg-config file now includes the OpenMP linker flag if necessary</li>
<li>New program ct2db allows for conversion of .ct files into dot-bracket notation (incl. pseudo-knot removal)</li>
</ul>
v 2.1.5</p><ul>
<li>Fix for discrepancy between special hairpin loop evaluation in partition functions and MFE</li>
</ul>
v 2.1.4</p><ul>
<li>Fix of G-quadruplex support in <a class="el" href="group__subopt__wuchty.html#ga700f662506a233e42dd7fda74fafd40e" title="Returns list of subopt structures or writes to fp. ">subopt()</a></li>
<li>Fix for discrepancy between special hairpin loop evaluation in partition functions and MFE</li>
</ul>
v 2.1.3</p><ul>
<li>RNAfold: Bugfix for ignoring user specified energy parameter files</li>
<li>RNAcofold: Bugfix for crashing upon constrained folding without specifying a constraint structure</li>
<li>RNAsubopt: Added G-quadruplex support</li>
<li>RNAalifold: Added parameter option to specify base pair probability threshold in dotplot</li>
<li>Fix of several G-quadruplex related bugs</li>
<li>Added G-quadruplex support in <a class="el" href="group__subopt__wuchty.html#ga700f662506a233e42dd7fda74fafd40e" title="Returns list of subopt structures or writes to fp. ">subopt()</a></li>
</ul>
v 2.1.2</p><ul>
<li>RNAfold: Bugfix for randomly missing probabilities in dot-plot during batch job execution</li>
<li>RNAeval: Bugfix for misinterpreted G-quadruplex containing sequences where the quadruplex starts at nucleotide 1</li>
<li>RNAsubopt: Slight changes to the output of stochastic backtracking and zuker subopt</li>
<li>Fix of some memory leaks</li>
<li>Bugfixes in <a class="el" href="group__subopt__zuker.html#ga0d5104e3ecf119d8eabd40aa5fe47f90" title="Compute Zuker type suboptimal structures. ">zukersubopt()</a>, <a class="el" href="group__pf__fold.html#ga1cc05aaa9b0e7df2d3887e98321c2030" title="Create a vrna_ep_t from a probability matrix. ">assign_plist_from_pr()</a></li>
<li>New threadsafe variants of putoutpU_prob*() for LPfold()</li>
<li>Provision of python2 interface support.</li>
</ul>
v 2.1.1</p><ul>
<li>Bugfix to restore backward compatibility with ViennaRNA Package 1.8.x API (this bug also affected proper usage of the the perl interface)</li>
</ul>
v 2.1.0</p><ul>
<li>G-Quadruplex support in RNAfold, RNAcofold, RNALfold, RNAalifold, RNAeval and RNAplot</li>
<li>LPfold got a new option to output its computations in split-mode</li>
<li>several G-Quadruplex related functions were introduced with this release</li>
<li>several functions for moves in an RNA landscape were introduced</li>
<li>new function in alipfold.c now enables access to the partition function matrices of <a class="el" href="group__consensus__pf__fold.html#ga1a5f6cfb9d761fa862ce4edc7c369cd2" title="The partition function version of alifold() works in analogy to pf_fold(). Pair probabilities and inf...">alipf_fold()</a></li>
<li>different numeric approach was implement for concentration dependend co-folding to avoid instabilities which occured under certain circumstances</li>
</ul>
v 2.0.7</p><ul>
<li>Bugfix for RNAplfold where segfault happened upon usage of -O option</li>
<li>Corrected misbehavior of RNAeval and RNAplot in tty mode</li>
</ul>
v 2.0.6</p><ul>
<li>Bugfix for bad type casting with gcc under MacOSX (resulted in accidental "sequence too long" errors)</li>
<li>Bugfix for disappearing tri-/hexaloop contributions when read in from certain parameter files</li>
<li>Bugfix for RNALfold that segfaulted on short strange sequences like AT+ repeats</li>
<li>Change of RNA2Dfold output format for stochastic backtracking</li>
</ul>
v 2.0.5</p><ul>
<li>Restored z-score computation capabilities in RNALfold</li>
</ul>
v 2.0.4</p><ul>
<li>Bugfix for RNAcofold partition function</li>
<li>Perl wrapper compatibility to changed RNAlib has been restored</li>
<li>Backward compatibility for partition function calls has been restored</li>
</ul>
v 2.0.3</p><ul>
<li>Bugfix for RNAalifold partition function and base pair probabilities in v2.0.3b</li>
<li>Added Boltzmann factor scaling in RNAsubopt, RNAalifold, RNAplfold and RNAcofold</li>
<li>Bugfix for alipfold() in v2.0.3b</li>
<li>Restored threadsafety of folding matrix access in LPfold.c, alipfold.c, part_func.c, part_func_co.c and part_func_up.c</li>
<li>Added several new functions regarding threadsafe function calls in terms of concurrently changing the model details</li>
<li>Added pkg-config file in the distribution to allow easy checks for certain RNAlib2 versions, compiler flags and linker flags.</li>
</ul>
v 2.0.2</p><ul>
<li>added support for Boltzmann factor scaling in RNAfold</li>
<li>fixed fastaheader to filename bug</li>
<li>plugged some memory leaks</li>
</ul>
v 2.0.1</p><ul>
<li>First official release of version 2.0</li>
<li>included latest bugfixes</li>
</ul>
__History__2011-03-10 Ronny Lorenz <a href="#" onclick="location.href='mai'+'lto:'+'ron'+'ny'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">ronny<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a></p><ul>
<li>new naming scheme for all shipped energy parameter files</li>
<li>fixed bugs that appear while compiling with gcc under MacOS X</li>
<li>fixed bug in RNAup --interaction-first where the longer of the first two sequences was taken as target</li>
<li>added full FASTA input support to RNAfold, RNAcofold, RNAheat, RNAplfold RNALfoldz, RNAsubopt and RNALfold</li>
</ul>
2010-11-24 Ronny Lorenz <a href="#" onclick="location.href='mai'+'lto:'+'ron'+'ny'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">ronny<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a></p><ul>
<li>first full pre-release of version 2.0</li>
</ul>
2009-11-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a></p><ul>
<li>Fix memory corruption in <a class="el" href="group__plotting__utils.html#ga821802c3685e37e15182341f6217470d" title="Produce PostScript sequence alignment color-annotated by consensus structure. ">PS_color_aln()</a></li>
</ul>
2009-09-09 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a></p><ul>
<li>Fix bug in RNAplfold when -u and -L parameters are equal</li>
<li>Fix double call to <a class="el" href="group__mfe__fold__single.html#ga107fdfe5fd641868156bfd849f6866c7" title="Free arrays for mfe folding. ">free_arrays()</a> in RNAfold.c</li>
<li>Improve drawing of cofolded structures</li>
</ul>
2009-05-14 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Fix occasional segfault in RNAalifold's print_aliout()</li>
</ul>
2009-02-24 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Add -MEA options to RNAfold and RNAalifold</li>
<li>change energy_of_alistruct to return float not void</li>
</ul>
2009-02-24 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>RNAfold will draw structures unless -noPS is used (no more "structure too long" messages)</li>
<li>Restore the "alifold.out" output from RNAalifold -p</li>
<li>RNAalifold -circ did not work due to wrong return type</li>
<li>Accessibility calculation with RNAplfold would give wrong results for u&lt;=30</li>
</ul>
2008-12-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Add zuker style suboptimals to RNAsubopt (-z)</li>
<li><a class="el" href="utils_8h.html#abe51806d14cff0789a8c1df7dbc45b71" title="Read a line of arbitrary length from a stream. ">get_line()</a> should be much faster when reading huge sequences (e.g. whole chromosomes for RNALfold)</li>
</ul>
2008-08-12 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Add Ribosum matrices for covariance scoring in RNAalifold</li>
</ul>
2008-06-27 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Change RNAalifold to used berni's new energy evaluation w/o gaps</li>
<li>Add stochastic backtracking in RNAalifold</li>
</ul>
2008-07-04 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>modify output of RNAup (again). Program reading RNAup output will have to updated!</li>
</ul>
2008-07-02 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>RNAplfold now computes accessibilities for all regions up to a max length simultaneously. Slightly slower when only 1 value is needed, but much faster if all of them are wanted. This entails a new output format. Programs reading accessibility output from RNAplfold need to be updated!</li>
</ul>
2008-03-31 Stephan Bernhart <a href="#" onclick="location.href='mai'+'lto:'+'ber'+'ni'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">berni<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a><ul>
<li>add cofolding to RNAsubopt</li>
</ul>
2008-01-08 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>ensure circfold works even for open chain</li>
</ul>
2007-12-13 Ulli Mueckstein <a href="#" onclick="location.href='mai'+'lto:'+'ull'+'i@'+'tbi'+'.u'+'niv'+'ie'+'.ac'+'.a'+'t'; return false;">ulli@<span style="display: none;">.nosp@m.</span>tbi.<span style="display: none;">.nosp@m.</span>univi<span style="display: none;">.nosp@m.</span>e.ac<span style="display: none;">.nosp@m.</span>.at</a><ul>
<li>upate RNAup related files RNAup can now include the intramolecular structure of both molecules and handles constraints.</li>
</ul>
2007-12-05 Ronny Lorenz <a href="#" onclick="location.href='mai'+'lto:'+'ron'+'ny'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">ronny<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a><ul>
<li>add circfold variants in part_func.c alipfold.c subopt.c</li>
</ul>
2007-09-19 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>compute the controid structure of the ensemble in RNAfold -p</li>
<li>fix a missing factor 2 in <a class="el" href="part__func_8h.html#ae9556ba7ded44fe2321b6f67c3fc02a3">mean_bp_dist()</a>. CAUTION ensemble diversities returned by RNAfold -p are now twice as large as in earlier versions.</li>
</ul>
2007-09-04 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@b'+'lin'+'i.'+'tbi'+'.u'+'niv'+'ie'+'.ac'+'.a'+'t'; return false;">ivo@b<span style="display: none;">.nosp@m.</span>lini<span style="display: none;">.nosp@m.</span>.tbi.<span style="display: none;">.nosp@m.</span>univ<span style="display: none;">.nosp@m.</span>ie.ac<span style="display: none;">.nosp@m.</span>.at</a><ul>
<li>fix a bug in <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a> where base number n-max-4 would never pair</li>
</ul>
2007-08-26 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add RNAaliduplex the alignment version of RNAduplex</li>
<li>introduce a minimal distance between hits produced by duplex_subopt()</li>
</ul>
2007-07-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add a <a class="el" href="group__eval.html#ga507d4fd93f4b398d793ba2402731388d" title="Calculate energy of a loop. ">loop_energy()</a> function to compute energy of a single loop</li>
</ul>
2007-06-23 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add <a class="el" href="group__local__consensus__fold.html#ga864db0f9c428356d978ad0600aa9a8bc">aliLfold()</a> and RNALalifold, alignment variant of <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a></li>
</ul>
2007-04-30 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a></p><ul>
<li>add RNAup to distribution</li>
</ul>
2007-04-15 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a></p><ul>
<li>fix segfault in colorps output (thanks to Andres Varon)</li>
</ul>
2007-03-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a></p><ul>
<li>avoid unnormalized doubles in scale[], big speedup for <a class="el" href="group__pf__fold.html#gadc3db3d98742427e7001a7fd36ef28c2" title="Compute the partition function  of an RNA sequence. ">pf_fold()</a> on very long sequences</li>
</ul>
2007-02-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>RNAalifold can now produce colored structure plots and alignment plots</li>
</ul>
2007-02-01 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Fix segfault in RNAplfold because of missing prototype</li>
</ul>
2006-12-01 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>RNAduplex would segfault when no structure base pairs are possible</li>
</ul>
2006-08-22 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add computation stacking probabilities using RNAfold -p2</li>
<li>add -noPS option for NRAfold to supress drawing structures</li>
</ul>
2006-08-09 Stephan Bernhart <a href="#" onclick="location.href='mai'+'lto:'+'ber'+'ni'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">berni<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a><ul>
<li>RNAplfold can now compute probabilites of unpaired regions (scanning version of RNAup)</li>
</ul>
2006-06-14 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>compile library with -fpic (if available) for use as shared library in the Perl module.</li>
<li>fix another bug when calling <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a> repeatedly</li>
<li>fix switch cmdline parsing in RNAalifold (-mis implied -4)</li>
<li>fix bug in <a class="el" href="group__mfe__cofold.html#gabc8517f22cfe70595ee81fc837910d52" title="Compute the minimum free energy of two interacting RNA molecules. ">cofold()</a> with dangles=0</li>
</ul>
2006-05-08 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>fix segfault in <a class="el" href="group__local__mfe__fold.html#ga16e5a70e60835bb969eaecbe6482f1be" title="The local analog to fold(). ">Lfold()</a> when calling repeatedly</li>
<li>fix structure parsing in RNAstruct.c (thanks to Michael Pheasant for reporting both bugs)</li>
<li>add duplexfold() and <a class="el" href="group__consensus__mfe__fold.html#ga4cf00f0659e5f0480335d69e797f05b1" title="Compute MFE and according consensus structure of an alignment of sequences. ">alifold()</a> to Perl module</li>
<li>distinguish window size and max pair span in LPfold</li>
</ul>
2006-04-05 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>fix performance bug in <a class="el" href="part__func__co_8h.html#ae5c1e7331718669bdae7a86de2be6184" title="Calculate partition function and base pair probabilities. ">co_pf_fold()</a></li>
<li>use relative error for termination of Newton iteration</li>
</ul>
2006-03-02 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add circular folding in <a class="el" href="group__consensus__mfe__fold.html#ga4cf00f0659e5f0480335d69e797f05b1" title="Compute MFE and according consensus structure of an alignment of sequences. ">alifold()</a></li>
</ul>
2006-01-18 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>cleanup berni partition cofold code, including several bug fixes</li>
</ul>
2006-01-16 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>update RNAplfold to working version</li>
<li>add PS_dot_plot_turn() in PS_dot.c</li>
</ul>
2005-11-07 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add new utilities colorna and coloraln</li>
</ul>
2005-10-11 Christoph Flamm <a href="#" onclick="location.href='mai'+'lto:'+'xto'+'f@'+'tbi'+'.u'+'niv'+'ie'+'.ac'+'.a'+'t'; return false;">xtof@<span style="display: none;">.nosp@m.</span>tbi.<span style="display: none;">.nosp@m.</span>univi<span style="display: none;">.nosp@m.</span>e.ac<span style="display: none;">.nosp@m.</span>.at</a><ul>
<li>adapt <a class="el" href="plot__structure_8h.html#a0873c7cc4cd7a11c9a2cea19dde7e9c9" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">PS_rna_plot()</a> for drawing co-folded structures</li>
</ul>
2005-07-24 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>fix a few memory problems in structure comparison routines</li>
</ul>
2005-04-30 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@b'+'lin'+'i.'+'tbi'+'.u'+'niv'+'ie'+'.ac'+'.a'+'t'; return false;">ivo@b<span style="display: none;">.nosp@m.</span>lini<span style="display: none;">.nosp@m.</span>.tbi.<span style="display: none;">.nosp@m.</span>univ<span style="display: none;">.nosp@m.</span>ie.ac<span style="display: none;">.nosp@m.</span>.at</a><ul>
<li>add folding of circular RNAs</li>
</ul>
2005-03-11 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@b'+'lin'+'i.'+'tbi'+'.u'+'niv'+'ie'+'.ac'+'.a'+'t'; return false;">ivo@b<span style="display: none;">.nosp@m.</span>lini<span style="display: none;">.nosp@m.</span>.tbi.<span style="display: none;">.nosp@m.</span>univ<span style="display: none;">.nosp@m.</span>ie.ac<span style="display: none;">.nosp@m.</span>.at</a><ul>
<li>add -mis option to RNAalifold to give "most informative
	  sequence" as consensus</li>
</ul>
2005-02-10 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>move <a class="el" href="group__consensus__mfe__fold.html#ga4cf00f0659e5f0480335d69e797f05b1" title="Compute MFE and according consensus structure of an alignment of sequences. ">alifold()</a> into the library</li>
</ul>
2004-12-22 Stephan Bernhart <a href="#" onclick="location.href='mai'+'lto:'+'ber'+'ni'+'@tb'+'i.'+'uni'+'vi'+'e.a'+'c.'+'at'; return false;">berni<span style="display: none;">.nosp@m.</span>@tbi<span style="display: none;">.nosp@m.</span>.univ<span style="display: none;">.nosp@m.</span>ie.a<span style="display: none;">.nosp@m.</span>c.at</a><ul>
<li>add partition function version of RNAcofold</li>
</ul>
2004-12-23 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add RNApaln for fast structural alignments (RNApdist improvement)</li>
</ul>
2004-08-12 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>fix constrained folding in stochastic backtracking</li>
</ul>
2004-07-21 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add RNAduplex, to compute hybrid structures without intra-molecular pairs</li>
</ul>
2004-02-09 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>fix bug in fold that caused segfaults when using Intel compiler</li>
<li>add computation of ensemble diversity to RNAfold</li>
</ul>
2003-09-10 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add annotation options to RNAplot</li>
</ul>
2003-08-04 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>stochastic backtracking finally works. Try e.g. RNAsubopt -p 10</li>
</ul>
2003-07-18 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add relplot.pl and rotate_ss.pl utilities for reliability annotation and rotation of rna structure plots</li>
</ul>
2003-01-29 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add RNALfold program to compute locally optimal structures with maximum pair span.</li>
<li>add RNAcofold for computing hybrid structure</li>
</ul>
2002-11-07 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>change <a class="el" href="profiledist_8h.html#a904c7eaf4a2413567c00ac4891749d18">Make_bp_profile()</a> and <a class="el" href="profiledist_8h.html#abe75e90e00a1e5dd8862944ed53dad5d" title="Align the 2 probability profiles T1, T2 . ">profile_edit_distance()</a> to use simple (float *) arrays; makes Perl access much easier. RNApdist -B now works again</li>
</ul>
2002-10-28 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Improved Perl module with pod documentation; allow to write things like ($structure, $energy) = RNA::fold($seq); Compatibility warning: the ptrvalue() and related functions are gone, see the pod documentation for alternatives.</li>
</ul>
2002-10-29 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>added svg structure plots in PS_dot.c and RNAplot</li>
</ul>
2002-08-15 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>Improve reading of clustal files (alifold)</li>
<li>add a sample alifold.cgi script</li>
</ul>
2001-09-18 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>moved suboptimal folding into the library, thus it's now accessible from the Perl module</li>
</ul>
2001-08-31 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>added co-folding support in <a class="el" href="group__eval.html#gac2b37fea2145c94d925a3f33378ef87b">energy_of_struct()</a>, and thus RNAeval</li>
</ul>
2001-04-30 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>switch from handcrafted makefiles to automake and autoconf</li>
</ul>
2001-04-05 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>added PS_rna_plot_a to produce structure plots with annotation</li>
</ul>
2001-03-03 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add alifold; predict consensus structures from alignment</li>
</ul>
2000-09-28 Ivo Hofacker <a href="#" onclick="location.href='mai'+'lto:'+'ivo'+'@t'+'bi.'+'un'+'ivi'+'e.'+'ac.'+'at'; return false;">ivo@t<span style="display: none;">.nosp@m.</span>bi.u<span style="display: none;">.nosp@m.</span>nivie<span style="display: none;">.nosp@m.</span>.ac.<span style="display: none;">.nosp@m.</span>at</a><ul>
<li>add -d3 option to RNAfold for co-axial stacking </li>
</ul>
 </p>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Thu Apr 19 2018 13:26:55 for RNAlib-2.4.6 by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
