<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>RNAlib-2.4.6: Functions for Creating Secondary Structure Plots, Dot-Plots, and More</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() {
    if ($('.searchresults').length > 0) { searchBox.DOMSearchField().focus(); }
  });
</script>
<link rel="search" href="search_opensearch.php?v=opensearch.xml" type="application/opensearchdescription+xml" title="RNAlib-2.4.6"/>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">RNAlib-2.4.6
   </div>
  </td>
   <td>        <div id="MSearchBox" class="MSearchBoxInactive">
          <div class="left">
            <form id="FSearchBox" action="search.html" method="get">
              <img id="MSearchSelect" src="search/mag.png" alt=""/>
              <input type="text" id="MSearchField" name="query" value="Search" size="20" accesskey="S" 
                     onfocus="searchBox.OnSearchFieldFocus(true)" 
                     onblur="searchBox.OnSearchFieldFocus(false)"/>
            </form>
          </div><div class="right"></div>
        </div>
</td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('group__plotting__utils.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="summary">
<a href="#groups">Modules</a> &#124;
<a href="#files">Files</a> &#124;
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#define-members">Macros</a> &#124;
<a href="#func-members">Functions</a> &#124;
<a href="#var-members">Variables</a>  </div>
  <div class="headertitle">
<div class="title">Functions for Creating Secondary Structure Plots, Dot-Plots, and More<div class="ingroups"><a class="el" href="group__utils.html">Utilities</a></div></div>  </div>
</div><!--header-->
<div class="contents">
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Collaboration diagram for Functions for Creating Secondary Structure Plots, Dot-Plots, and More:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<center><table><tr><td><div class="center"><iframe scrolling="no" frameborder="0" src="group__plotting__utils.svg" width="536" height="82"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</td></tr></table></center>
</div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="groups"></a>
Modules</h2></td></tr>
<tr class="memitem:group__annotation__utils"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__annotation__utils.html">Functions to generate annotations for Secondary Structure Plots, Dot-Plots, and Others</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="files"></a>
Files</h2></td></tr>
<tr class="memitem:naview_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="naview_8h.html">naview.h</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:plot__aln_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="plot__aln_8h.html">plot_aln.h</a></td></tr>
<tr class="memdesc:plot__aln_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Various functions for plotting Sequence / Structure Alignments. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:plot__layouts_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="plot__layouts_8h.html">plot_layouts.h</a></td></tr>
<tr class="memdesc:plot__layouts_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Secondary structure plot layout algorithms. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:plot__structure_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="plot__structure_8h.html">plot_structure.h</a></td></tr>
<tr class="memdesc:plot__structure_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Various functions for plotting RNA secondary structures. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:plot__utils_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="plot__utils_8h.html">plot_utils.h</a></td></tr>
<tr class="memdesc:plot__utils_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Various utilities to assist in plotting secondary structures and consensus structures. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:PS__dot_8h"><td class="memItemLeft" align="right" valign="top">file &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="PS__dot_8h.html">PS_dot.h</a></td></tr>
<tr class="memdesc:PS__dot_8h"><td class="mdescLeft">&#160;</td><td class="mdescRight">Various functions for plotting RNA secondary structures, dot-plots and other visualizations. <br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:structCOORDINATE"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#structCOORDINATE">COORDINATE</a></td></tr>
<tr class="memdesc:structCOORDINATE"><td class="mdescLeft">&#160;</td><td class="mdescRight">this is a workarround for the SWIG Perl Wrapper RNA plot function that returns an array of type <a class="el" href="group__plotting__utils.html#structCOORDINATE" title="this is a workarround for the SWIG Perl Wrapper RNA plot function that returns an array of type COORD...">COORDINATE</a>  <a href="group__plotting__utils.html#structCOORDINATE">More...</a><br /></td></tr>
<tr class="separator:structCOORDINATE"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:structvrna__dotplot__auxdata__t"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#structvrna__dotplot__auxdata__t">vrna_dotplot_auxdata_t</a></td></tr>
<tr class="separator:structvrna__dotplot__auxdata__t"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="define-members"></a>
Macros</h2></td></tr>
<tr class="memitem:gae6d17b9f0a53cf5205a9181e0f8422e9"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#gae6d17b9f0a53cf5205a9181e0f8422e9">VRNA_PLOT_TYPE_SIMPLE</a>&#160;&#160;&#160;0</td></tr>
<tr class="memdesc:gae6d17b9f0a53cf5205a9181e0f8422e9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Definition of Plot type <em>simple</em>  <a href="#gae6d17b9f0a53cf5205a9181e0f8422e9">More...</a><br /></td></tr>
<tr class="separator:gae6d17b9f0a53cf5205a9181e0f8422e9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga94d4c863ecac2f220f76658afb92f964"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga94d4c863ecac2f220f76658afb92f964">VRNA_PLOT_TYPE_NAVIEW</a>&#160;&#160;&#160;1</td></tr>
<tr class="memdesc:ga94d4c863ecac2f220f76658afb92f964"><td class="mdescLeft">&#160;</td><td class="mdescRight">Definition of Plot type <em>Naview</em>  <a href="#ga94d4c863ecac2f220f76658afb92f964">More...</a><br /></td></tr>
<tr class="separator:ga94d4c863ecac2f220f76658afb92f964"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga8c9eac631348da92136c8363ecdd9fb9"><td class="memItemLeft" align="right" valign="top">#define&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga8c9eac631348da92136c8363ecdd9fb9">VRNA_PLOT_TYPE_CIRCULAR</a>&#160;&#160;&#160;2</td></tr>
<tr class="memdesc:ga8c9eac631348da92136c8363ecdd9fb9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Definition of Plot type <em>Circular</em>  <a href="#ga8c9eac631348da92136c8363ecdd9fb9">More...</a><br /></td></tr>
<tr class="separator:ga8c9eac631348da92136c8363ecdd9fb9"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:ga821802c3685e37e15182341f6217470d"><td class="memItemLeft" align="right" valign="top"><a id="ga821802c3685e37e15182341f6217470d"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga821802c3685e37e15182341f6217470d">PS_color_aln</a> (const char *structure, const char *filename, const char *seqs[], const char *names[])</td></tr>
<tr class="memdesc:ga821802c3685e37e15182341f6217470d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Produce PostScript sequence alignment color-annotated by consensus structure. <br /></td></tr>
<tr class="separator:ga821802c3685e37e15182341f6217470d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gaab48d4dac655d688abe921389ac2847c"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#gaab48d4dac655d688abe921389ac2847c">aliPS_color_aln</a> (const char *structure, const char *filename, const char *seqs[], const char *names[])</td></tr>
<tr class="separator:gaab48d4dac655d688abe921389ac2847c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gaf4b9173e7d3fd361c3c85e6def194123"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#gaf4b9173e7d3fd361c3c85e6def194123">simple_xy_coordinates</a> (short *pair_table, float *X, float *Y)</td></tr>
<tr class="memdesc:gaf4b9173e7d3fd361c3c85e6def194123"><td class="mdescLeft">&#160;</td><td class="mdescRight">Calculate nucleotide coordinates for secondary structure plot the <em>Simple way</em>  <a href="#gaf4b9173e7d3fd361c3c85e6def194123">More...</a><br /></td></tr>
<tr class="separator:gaf4b9173e7d3fd361c3c85e6def194123"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:gac4ea13d35308f09940178d2b05a248c2"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#gac4ea13d35308f09940178d2b05a248c2">simple_circplot_coordinates</a> (short *pair_table, float *x, float *y)</td></tr>
<tr class="memdesc:gac4ea13d35308f09940178d2b05a248c2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Calculate nucleotide coordinates for <em>Circular Plot</em>  <a href="#gac4ea13d35308f09940178d2b05a248c2">More...</a><br /></td></tr>
<tr class="separator:gac4ea13d35308f09940178d2b05a248c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga00ea223b5cf02eb2faae5ff29f0d5e12"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga00ea223b5cf02eb2faae5ff29f0d5e12">PS_dot_plot_list</a> (char *seq, char *filename, <a class="el" href="group__data__structures.html#ga9608eed021ebfbdd7a901cfdc446c8e9">plist</a> *pl, <a class="el" href="group__data__structures.html#ga9608eed021ebfbdd7a901cfdc446c8e9">plist</a> *mf, char *comment)</td></tr>
<tr class="memdesc:ga00ea223b5cf02eb2faae5ff29f0d5e12"><td class="mdescLeft">&#160;</td><td class="mdescRight">Produce a postscript dot-plot from two pair lists.  <a href="#ga00ea223b5cf02eb2faae5ff29f0d5e12">More...</a><br /></td></tr>
<tr class="separator:ga00ea223b5cf02eb2faae5ff29f0d5e12"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ga689a97a7e3b8a2df14728b8204d9d57b"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga689a97a7e3b8a2df14728b8204d9d57b">PS_dot_plot</a> (char *string, char *file)</td></tr>
<tr class="memdesc:ga689a97a7e3b8a2df14728b8204d9d57b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Produce postscript dot-plot.  <a href="#ga689a97a7e3b8a2df14728b8204d9d57b">More...</a><br /></td></tr>
<tr class="separator:ga689a97a7e3b8a2df14728b8204d9d57b"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="var-members"></a>
Variables</h2></td></tr>
<tr class="memitem:ga5964c4581431b098b80027d6e14dcdd4"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4">rna_plot_type</a></td></tr>
<tr class="memdesc:ga5964c4581431b098b80027d6e14dcdd4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Switch for changing the secondary structure layout algorithm.  <a href="#ga5964c4581431b098b80027d6e14dcdd4">More...</a><br /></td></tr>
<tr class="separator:ga5964c4581431b098b80027d6e14dcdd4"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<hr/><h2 class="groupheader">Data Structure Documentation</h2>
<a name="structCOORDINATE" id="structCOORDINATE"></a>
<h2 class="memtitle"><span class="permalink"><a href="#structCOORDINATE">&#9670;&nbsp;</a></span>COORDINATE</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">struct COORDINATE</td>
        </tr>
      </table>
</div><div class="memdoc">
<div class="textblock"><p>this is a workarround for the SWIG Perl Wrapper RNA plot function that returns an array of type <a class="el" href="group__plotting__utils.html#structCOORDINATE" title="this is a workarround for the SWIG Perl Wrapper RNA plot function that returns an array of type COORD...">COORDINATE</a> </p>
</div>
</div>
</div>
<a name="structvrna__dotplot__auxdata__t" id="structvrna__dotplot__auxdata__t"></a>
<h2 class="memtitle"><span class="permalink"><a href="#structvrna__dotplot__auxdata__t">&#9670;&nbsp;</a></span>vrna_dotplot_auxdata_t</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">struct vrna_dotplot_auxdata_t</td>
        </tr>
      </table>
</div><div class="memdoc">
<div class="textblock"></div><div id="dynsection-1" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-1-trigger" src="closed.png" alt="+"/> Collaboration diagram for vrna_dotplot_auxdata_t:</div>
<div id="dynsection-1-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-1-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="structvrna__dotplot__auxdata__t__coll__graph.svg" width="168" height="260"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
<center><span class="legend">[<a target="top" href="graph_legend.html">legend</a>]</span></center></div>

</div>
</div>
<h2 class="groupheader">Macro Definition Documentation</h2>
<a id="gae6d17b9f0a53cf5205a9181e0f8422e9"></a>
<h2 class="memtitle"><span class="permalink"><a href="#gae6d17b9f0a53cf5205a9181e0f8422e9">&#9670;&nbsp;</a></span>VRNA_PLOT_TYPE_SIMPLE</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_PLOT_TYPE_SIMPLE&#160;&#160;&#160;0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Definition of Plot type <em>simple</em> </p>
<p>This is the plot type definition for several RNA structure plotting functions telling them to use <b>Simple</b> plotting algorithm</p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4" title="Switch for changing the secondary structure layout algorithm. ">rna_plot_type</a>, <a class="el" href="plot__structure_8h.html#a139a31dd0ba9fc6612431f67de901c31" title="Produce a secondary structure graph in PostScript including additional annotation macros and write it...">vrna_file_PS_rnaplot_a()</a>, <a class="el" href="plot__structure_8h.html#abdc8f6548ba4a3bc3cd868ccbcfdb86a" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">vrna_file_PS_rnaplot()</a>, <a class="el" href="plot__structure_8h.html#ae7853539b5df98f294b4af434e979304" title="Produce a secondary structure plot in SVG format and write it to a file. ">svg_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a70834bc8c0aad4fe6824ff76ccb8f329" title="Produce a secondary structure graph in Graph Meta Language (gml) and write it to a file...">gmlRNA()</a>, <a class="el" href="plot__structure_8h.html#add368528755f9a830727b680243541df" title="Produce a secondary structure graph in SStructView format. ">ssv_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a2f6d5953e6a323df898896b8d6614483" title="Produce a secondary structure plot for further editing in XRNA. ">xrna_plot()</a> </dd></dl>

</div>
</div>
<a id="ga94d4c863ecac2f220f76658afb92f964"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ga94d4c863ecac2f220f76658afb92f964">&#9670;&nbsp;</a></span>VRNA_PLOT_TYPE_NAVIEW</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_PLOT_TYPE_NAVIEW&#160;&#160;&#160;1</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Definition of Plot type <em>Naview</em> </p>
<p>This is the plot type definition for several RNA structure plotting functions telling them to use <b>Naview</b> plotting algorithm</p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4" title="Switch for changing the secondary structure layout algorithm. ">rna_plot_type</a>, <a class="el" href="plot__structure_8h.html#a139a31dd0ba9fc6612431f67de901c31" title="Produce a secondary structure graph in PostScript including additional annotation macros and write it...">vrna_file_PS_rnaplot_a()</a>, <a class="el" href="plot__structure_8h.html#abdc8f6548ba4a3bc3cd868ccbcfdb86a" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">vrna_file_PS_rnaplot()</a>, <a class="el" href="plot__structure_8h.html#ae7853539b5df98f294b4af434e979304" title="Produce a secondary structure plot in SVG format and write it to a file. ">svg_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a70834bc8c0aad4fe6824ff76ccb8f329" title="Produce a secondary structure graph in Graph Meta Language (gml) and write it to a file...">gmlRNA()</a>, <a class="el" href="plot__structure_8h.html#add368528755f9a830727b680243541df" title="Produce a secondary structure graph in SStructView format. ">ssv_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a2f6d5953e6a323df898896b8d6614483" title="Produce a secondary structure plot for further editing in XRNA. ">xrna_plot()</a> </dd></dl>

</div>
</div>
<a id="ga8c9eac631348da92136c8363ecdd9fb9"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ga8c9eac631348da92136c8363ecdd9fb9">&#9670;&nbsp;</a></span>VRNA_PLOT_TYPE_CIRCULAR</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">#define VRNA_PLOT_TYPE_CIRCULAR&#160;&#160;&#160;2</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Definition of Plot type <em>Circular</em> </p>
<p>This is the plot type definition for several RNA structure plotting functions telling them to produce a <b>Circular plot</b></p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4" title="Switch for changing the secondary structure layout algorithm. ">rna_plot_type</a>, <a class="el" href="plot__structure_8h.html#a139a31dd0ba9fc6612431f67de901c31" title="Produce a secondary structure graph in PostScript including additional annotation macros and write it...">vrna_file_PS_rnaplot_a()</a>, <a class="el" href="plot__structure_8h.html#abdc8f6548ba4a3bc3cd868ccbcfdb86a" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">vrna_file_PS_rnaplot()</a>, <a class="el" href="plot__structure_8h.html#ae7853539b5df98f294b4af434e979304" title="Produce a secondary structure plot in SVG format and write it to a file. ">svg_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a70834bc8c0aad4fe6824ff76ccb8f329" title="Produce a secondary structure graph in Graph Meta Language (gml) and write it to a file...">gmlRNA()</a>, <a class="el" href="plot__structure_8h.html#add368528755f9a830727b680243541df" title="Produce a secondary structure graph in SStructView format. ">ssv_rna_plot()</a>, <a class="el" href="plot__structure_8h.html#a2f6d5953e6a323df898896b8d6614483" title="Produce a secondary structure plot for further editing in XRNA. ">xrna_plot()</a> </dd></dl>

</div>
</div>
<h2 class="groupheader">Function Documentation</h2>
<a id="gaab48d4dac655d688abe921389ac2847c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#gaab48d4dac655d688abe921389ac2847c">&#9670;&nbsp;</a></span>aliPS_color_aln()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int aliPS_color_aln </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>structure</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>filename</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>seqs</em>[], </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>names</em>[]&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__aln_8h.html">ViennaRNA/plot_aln.h</a>&gt;</code></p>
<p>PS_color_aln for duplexes </p>

</div>
</div>
<a id="gaf4b9173e7d3fd361c3c85e6def194123"></a>
<h2 class="memtitle"><span class="permalink"><a href="#gaf4b9173e7d3fd361c3c85e6def194123">&#9670;&nbsp;</a></span>simple_xy_coordinates()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int simple_xy_coordinates </td>
          <td>(</td>
          <td class="paramtype">short *&#160;</td>
          <td class="paramname"><em>pair_table</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float *&#160;</td>
          <td class="paramname"><em>X</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float *&#160;</td>
          <td class="paramname"><em>Y</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Calculate nucleotide coordinates for secondary structure plot the <em>Simple way</em> </p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__struct__utils.html#ga89c32307ee50a0026f4a3131fac0845a" title="Create a pair table of a secondary structure. ">make_pair_table()</a>, <a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4" title="Switch for changing the secondary structure layout algorithm. ">rna_plot_type</a>, <a class="el" href="group__plotting__utils.html#gac4ea13d35308f09940178d2b05a248c2" title="Calculate nucleotide coordinates for Circular Plot ">simple_circplot_coordinates()</a>, naview_xy_coordinates(), <a class="el" href="plot__structure_8h.html#a139a31dd0ba9fc6612431f67de901c31" title="Produce a secondary structure graph in PostScript including additional annotation macros and write it...">vrna_file_PS_rnaplot_a()</a>, <a class="el" href="plot__structure_8h.html#abdc8f6548ba4a3bc3cd868ccbcfdb86a" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">vrna_file_PS_rnaplot</a>, <a class="el" href="plot__structure_8h.html#ae7853539b5df98f294b4af434e979304" title="Produce a secondary structure plot in SVG format and write it to a file. ">svg_rna_plot()</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">pair_table</td><td>The pair table of the secondary structure </td></tr>
    <tr><td class="paramname">X</td><td>a pointer to an array with enough allocated space to hold the x coordinates </td></tr>
    <tr><td class="paramname">Y</td><td>a pointer to an array with enough allocated space to hold the y coordinates </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>length of sequence on success, 0 otherwise </dd></dl>

</div>
</div>
<a id="gac4ea13d35308f09940178d2b05a248c2"></a>
<h2 class="memtitle"><span class="permalink"><a href="#gac4ea13d35308f09940178d2b05a248c2">&#9670;&nbsp;</a></span>simple_circplot_coordinates()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int simple_circplot_coordinates </td>
          <td>(</td>
          <td class="paramtype">short *&#160;</td>
          <td class="paramname"><em>pair_table</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float *&#160;</td>
          <td class="paramname"><em>x</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">float *&#160;</td>
          <td class="paramname"><em>y</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Calculate nucleotide coordinates for <em>Circular Plot</em> </p>
<p>This function calculates the coordinates of nucleotides mapped in equal distancies onto a unit circle.</p>
<dl class="section note"><dt>Note</dt><dd>In order to draw nice arcs using quadratic bezier curves that connect base pairs one may calculate a second tangential point <img class="formulaInl" alt="$P^t$" src="form_108.png"/> in addition to the actual R<sup>2</sup> coordinates. the simplest way to do so may be to compute a radius scaling factor <img class="formulaInl" alt="$rs$" src="form_109.png"/> in the interval <img class="formulaInl" alt="$[0,1]$" src="form_110.png"/> that weights the proportion of base pair span to the actual length of the sequence. This scaling factor can then be used to calculate the coordinates for <img class="formulaInl" alt="$P^t$" src="form_108.png"/>, i.e. <img class="formulaInl" alt="$ P^{t}_x[i] = X[i] * rs$" src="form_111.png"/> and <img class="formulaInl" alt="$P^{t}_y[i] = Y[i] * rs$" src="form_112.png"/>.</dd></dl>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__struct__utils.html#ga89c32307ee50a0026f4a3131fac0845a" title="Create a pair table of a secondary structure. ">make_pair_table()</a>, <a class="el" href="group__plotting__utils.html#ga5964c4581431b098b80027d6e14dcdd4" title="Switch for changing the secondary structure layout algorithm. ">rna_plot_type</a>, <a class="el" href="group__plotting__utils.html#gaf4b9173e7d3fd361c3c85e6def194123" title="Calculate nucleotide coordinates for secondary structure plot the Simple way ">simple_xy_coordinates()</a>, naview_xy_coordinates(), <a class="el" href="plot__structure_8h.html#a139a31dd0ba9fc6612431f67de901c31" title="Produce a secondary structure graph in PostScript including additional annotation macros and write it...">vrna_file_PS_rnaplot_a()</a>, <a class="el" href="plot__structure_8h.html#abdc8f6548ba4a3bc3cd868ccbcfdb86a" title="Produce a secondary structure graph in PostScript and write it to &#39;filename&#39;. ">vrna_file_PS_rnaplot</a>, <a class="el" href="plot__structure_8h.html#ae7853539b5df98f294b4af434e979304" title="Produce a secondary structure plot in SVG format and write it to a file. ">svg_rna_plot()</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">pair_table</td><td>The pair table of the secondary structure </td></tr>
    <tr><td class="paramname">x</td><td>a pointer to an array with enough allocated space to hold the x coordinates </td></tr>
    <tr><td class="paramname">y</td><td>a pointer to an array with enough allocated space to hold the y coordinates </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>length of sequence on success, 0 otherwise </dd></dl>

</div>
</div>
<a id="ga00ea223b5cf02eb2faae5ff29f0d5e12"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ga00ea223b5cf02eb2faae5ff29f0d5e12">&#9670;&nbsp;</a></span>PS_dot_plot_list()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int PS_dot_plot_list </td>
          <td>(</td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>filename</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__data__structures.html#ga9608eed021ebfbdd7a901cfdc446c8e9">plist</a> *&#160;</td>
          <td class="paramname"><em>pl</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="group__data__structures.html#ga9608eed021ebfbdd7a901cfdc446c8e9">plist</a> *&#160;</td>
          <td class="paramname"><em>mf</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>comment</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="PS__dot_8h.html">ViennaRNA/PS_dot.h</a>&gt;</code></p>

<p>Produce a postscript dot-plot from two pair lists. </p>
<p>This function reads two plist structures (e.g. base pair probabilities and a secondary structure) as produced by <a class="el" href="group__pf__fold.html#ga1cc05aaa9b0e7df2d3887e98321c2030" title="Create a vrna_ep_t from a probability matrix. ">assign_plist_from_pr()</a> and <a class="el" href="group__struct__utils.html#gab61df77cf7949cd516181fce0c3d7d78" title="Create a vrna_ep_t from a dot-bracket string. ">assign_plist_from_db()</a> and produces a postscript "dot plot" that is written to 'filename'.<br />
Using base pair probabilities in the first and mfe structure in the second plist, the resulting "dot plot" represents each base pairing probability by a square of corresponding area in a upper triangle matrix. The lower part of the matrix contains the minimum free energy structure.</p>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__pf__fold.html#ga1cc05aaa9b0e7df2d3887e98321c2030" title="Create a vrna_ep_t from a probability matrix. ">assign_plist_from_pr()</a>, <a class="el" href="group__struct__utils.html#gab61df77cf7949cd516181fce0c3d7d78" title="Create a vrna_ep_t from a dot-bracket string. ">assign_plist_from_db()</a></dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">seq</td><td>The RNA sequence </td></tr>
    <tr><td class="paramname">filename</td><td>A filename for the postscript output </td></tr>
    <tr><td class="paramname">pl</td><td>The base pair probability pairlist </td></tr>
    <tr><td class="paramname">mf</td><td>The mfe secondary structure pairlist </td></tr>
    <tr><td class="paramname">comment</td><td>A comment </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>1 if postscript was successfully written, 0 otherwise </dd></dl>

</div>
</div>
<a id="ga689a97a7e3b8a2df14728b8204d9d57b"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ga689a97a7e3b8a2df14728b8204d9d57b">&#9670;&nbsp;</a></span>PS_dot_plot()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int PS_dot_plot </td>
          <td>(</td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>string</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char *&#160;</td>
          <td class="paramname"><em>file</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="PS__dot_8h.html">ViennaRNA/PS_dot.h</a>&gt;</code></p>

<p>Produce postscript dot-plot. </p>
<p>Wrapper to PS_dot_plot_list</p>
<p>Reads base pair probabilities produced by <a class="el" href="group__pf__fold.html#gadc3db3d98742427e7001a7fd36ef28c2" title="Compute the partition function  of an RNA sequence. ">pf_fold()</a> from the global array <a class="el" href="fold__vars_8h.html#ac98ec419070aee6831b44e5c700f090f" title="A pointer to the base pair probability matrix. ">pr</a> and the pair list <a class="el" href="fold__vars_8h.html#a0244a629b5ab4f58b77590c3dfd130dc" title="Contains a list of base pairs after a call to fold(). ">base_pair</a> produced by <a class="el" href="group__mfe__fold__single.html#gaadafcb0f140795ae62e5ca027e335a9b" title="Compute minimum free energy and an appropriate secondary structure of an RNA sequence. ">fold()</a> and produces a postscript "dot plot" that is written to 'filename'. The "dot plot" represents each base pairing probability by a square of corresponding area in a upper triangle matrix. The lower part of the matrix contains the minimum free energy </p><dl class="section note"><dt>Note</dt><dd>DO NOT USE THIS FUNCTION ANYMORE SINCE IT IS NOT THREADSAFE</dd></dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000131">Deprecated:</a></b></dt><dd>This function is deprecated and will be removed soon! Use <a class="el" href="group__plotting__utils.html#ga00ea223b5cf02eb2faae5ff29f0d5e12">PS_dot_plot_list()</a> instead! </dd></dl>

</div>
</div>
<h2 class="groupheader">Variable Documentation</h2>
<a id="ga5964c4581431b098b80027d6e14dcdd4"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ga5964c4581431b098b80027d6e14dcdd4">&#9670;&nbsp;</a></span>rna_plot_type</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int rna_plot_type</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><code>#include &lt;<a class="el" href="plot__layouts_8h.html">ViennaRNA/plot_layouts.h</a>&gt;</code></p>

<p>Switch for changing the secondary structure layout algorithm. </p>
<p>Current possibility are 0 for a simple radial drawing or 1 for the modified radial drawing taken from the <em>naview</em> program of <a class="el" href="citelist.html#CITEREF_bruccoleri:1988">[3]</a>.</p>
<dl class="section note"><dt>Note</dt><dd>To provide thread safety please do not rely on this global variable in future implementations but pass a plot type flag directly to the function that decides which layout algorithm it may use!</dd></dl>
<dl class="section see"><dt>See also</dt><dd><a class="el" href="group__plotting__utils.html#gae6d17b9f0a53cf5205a9181e0f8422e9" title="Definition of Plot type simple ">VRNA_PLOT_TYPE_SIMPLE</a>, <a class="el" href="group__plotting__utils.html#ga94d4c863ecac2f220f76658afb92f964" title="Definition of Plot type Naview ">VRNA_PLOT_TYPE_NAVIEW</a>, <a class="el" href="group__plotting__utils.html#ga8c9eac631348da92136c8363ecdd9fb9" title="Definition of Plot type Circular ">VRNA_PLOT_TYPE_CIRCULAR</a> </dd></dl>

</div>
</div>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Thu Apr 19 2018 13:27:00 for RNAlib-2.4.6 by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
